dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java

```diff
@@ -42,6 +49,21 @@
             .put(PageMode.NAVIGATE_EDIT_MODE, VelocityNavigateEditMode::new)
             .build();
 
+    public VelocityModeHandler(
+            final HttpServletRequest request,
+            final HttpServletResponse response,
+            final IHTMLPage htmlPage,
+            final Host host) {
+        this.request = request;
+        this.response = response;
+        this.htmlPage = htmlPage;
+        this.host = host;
+    }
+
+    public void setMode(PageMode mode) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'mode' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4748140)